.pipelines/stages/jobs/py-publishing-job.yml

```diff
@@ -0,0 +1,161 @@
+parameters:
+- name: SpecificArtifact
+  displayName: Use Specific Artifact
+  type: boolean
+  default: false
+- name: BuildId
+  displayName: Specific Artifact's BuildId
+  type: string
+  default: '0'
+- name: arch
+  type: string
+- name: ep
+  type: string
+- name: cuda_version
+  type: string
+  default: ''
+- name: cuda_display_version
+  type: string
+  default: ''
+- name: os
+  type: string
+  values:
+  - 'linux'
+  - 'win'
+  - 'osx'
+- name: artifact_feed
+  type: string
+  default: 'ORT-Nightly'
+
+jobs:
+- job: python_${{ parameters.os }}_${{ parameters.ep }}${{ parameters.cuda_display_version }}_${{ parameters.arch }}_publishing
+  ${{ if eq(parameters.os, 'linux') }}:
+    pool:
+      ${{ if eq(parameters.arch, 'x64') }}:
+        name: 'onnxruntime-Ubuntu2204-AMD-CPU'
+      ${{ elseif eq(parameters.arch, 'arm64') }}:
+        name: 'onnxruntime-genai-Ubuntu2004-ARM-CPU'
+      ${{ else }}:
+        name: 'onnxruntime-Ubuntu2204-AMD-CPU'
+      os: linux
+  ${{ if eq(parameters.os, 'win') }}:
+    pool:
+      name: 'onnxruntime-Win-CPU-2022'
+      os: windows
+  ${{ if eq(parameters.os, 'osx') }}:
+    pool:
+      name: 'Azure Pipelines'
+      image: 'macOS-latest'
+      os: 'macOS'
+
+  strategy:
```

Comment:
I think you may do the publishing in a single job. You do not need to have a matrix here.  You can upload a macOS wheel from a windows machine.  You can use python 3.13 to upload wheels for any python version.